htmresearch/frameworks/nlp/classify_document_fingerprint.py

```diff
@@ -251,18 +258,43 @@ def classifyText(self, token, reset=0):
                             sequenceId=-1, reset=0)
       self.network.run(1)
 
-      if reset == 1:
-        self.reset()
-
       if self.verbosity >= 2:
         print "Classifying document:",document
         self.printRegionOutputs()
 
       self.currentDocument = None
-      return self.classifierRegion.getOutputData("categoriesOut")[0:self.numLabels]
+      categoryVotes = self.classifierRegion.getOutputData(
+          "categoriesOut")[0:self.numLabels]
+
+      if reset == 1:
+        self.reset()
+
+      if returnDetailedResults:
+        # Accumulate the ids. Sort results if requested
+        dist = self.classifierRegion.getSelf().getLatestDistances()
+
+        classifier = self.getClassifier()
+        if sortResults:
+          idList = []
+          sortedIndices = dist.argsort()
+          for i in sortedIndices:
+            idList.append(classifier.getPartitionId(i))
+          sortedDistances = dist[sortedIndices]
+          return categoryVotes, idList, sortedDistances
+
+        else:
+          idList = []
+          for i in range(len(dist)):
+            idList.append(classifier.getPartitionId(i))
```

Comment:
Replace this loop w/ `idList = [classifier.getPartitionId(i) for i in xrange(len(dist))]`.